orttraining/orttraining/python/ort_trainer.py

```diff
@@ -845,6 +879,63 @@ def _prepare_input_and_fetches(self, input_desc_with_, internal_learning_rate, i
 
         return input, fetches
 
+    def _compare_inference_output_pytorch_onnx(self, input, input_descs, output_desc):
+        # handle case if onnx model is provided instead of torch
+        if not self.torch_model_:
+            return
+
+        # set up training session for ort models (for inference)
+        no_postprocess_run_options = ort.RunOptions()
+        no_postprocess_run_options.only_execute_path_to_fetches = True
+        no_postprocess_run_options.training_mode = False
+        no_postprocess_output_desc = output_desc
+        
+        # handles case in which user specifies no post processing
+        postprocess_session_results = ort_training_session_run_helper(self.session, self.train_io_binding, input,
+                                                              input_descs, output_desc,
+                                                              self.device_,
+                                                              no_postprocess_run_options)
+        
+        no_postprocess_session_results = {}
+        if self._no_postprocess_onnx_model:
+            no_postprocess_session_results= ort_training_session_run_helper(self._no_postprocess_session, self._no_postprocess_train_io_binding, input,
+                                                                  input_descs, no_postprocess_output_desc,
+                                                                  self.device_,
+                                                                  no_postprocess_run_options)
+        else:
+            no_postprocess_session_results = postprocess_session_results # filler values not used
+
+        input_names = [elem.name_ for elem in input_descs]
+        cpu_data_list = []
+        from inspect import signature
+        # trailing inputs for ort optimizer are avoided by considering pytorch inputs directly
+        for i, (name, input_elem) in enumerate(zip(input_names, input)):
+            if i >= len(signature(self.torch_model_.forward).parameters): break
+            cpu_data_list.append(input_elem.to('cpu'))
+        cpu_data = tuple(cpu_data_list)
+
+        # torch inference step
+        with torch.no_grad():
+            torch_out = [self.torch_model_(*cpu_data)]
+        
+        # iterate through outputs, except loss when loss_fn is distinct from the model
+        if self.loss_fn_:
+            torch_out.insert(0, torch.empty(1, 1))
+        for ((out_name, npp_out), pp_out, t_out) in zip(no_postprocess_session_results.items(), postprocess_session_results.values(), torch_out):
+            if out_name == 'loss' and self.loss_fn_:
+                continue
+            no_postprocess_out = npp_out.cpu().numpy().flatten()
+            postprocess_out = pp_out.cpu().numpy().flatten()
+            t_out = t_out.cpu().numpy().flatten()
+            print("Comparing Values of {}".format(out_name))
```

Comment:
I was expecting that both `no_postprocess_out` and `postprocess_out` to be compared t_out and that the script would error out when it is higher than a certain tolerance (e.g. 0.01)